src/dtos/tile_color.rs

```diff
@@ -0,0 +1,277 @@
+use crate::traits::load_from::LoadFrom;
+use bevy::{asset::LoadContext, prelude::*};
+use serde::{de::Error, Deserialize, Deserializer};
+use std::{fmt::Display, str::Chars};
+
+#[derive(Debug, PartialEq)]
+pub struct TileColor {
+	color: Color,
+}
+
+impl TileColor {
+	fn from_parsed(_Parsed((_, Red(r), Green(g), Blue(b), Alpha(a), _)): _Parsed<All>) -> Self {
+		let color = Color::Srgba(Srgba {
+			red: r as f32 / 255.,
+			green: g as f32 / 255.,
+			blue: b as f32 / 255.,
+			alpha: a as f32 / 255.,
+		});
+
+		Self { color }
+	}
+}
+
+impl<'a> Deserialize<'a> for TileColor {
+	fn deserialize<TDeserializer>(deserializer: TDeserializer) -> Result<Self, TDeserializer::Error>
+	where
+		TDeserializer: Deserializer<'a>,
+	{
+		_TileColorString::deserialize(deserializer)?
+			.parse_color()
+			.map(TileColor::from_parsed)
+			.map_err(_Error::convert_to_error)
+	}
+}
+
+impl LoadFrom<TileColor> for ColorMaterial {
+	fn load_from(TileColor { color }: TileColor, _: &mut LoadContext) -> Self {
+		ColorMaterial::from_color(color)
+	}
+}
+
+#[derive(Debug, PartialEq, Deserialize)]
+/// A struct responsible of parsing hex encoded colors `#rrggbb` or `#rrggbbaa`
+/// to a bevy color for [`TileColor`]
+///
+///
+/// Let's be honest, the whole parsing process is hugely over-designed
+/// and could have been done with way fewer lines of code.
+///
+/// But what is life, if we are not enjoying ourselves?
+/// So scroll down and bask in the beauty of my nightly escapades :D
+struct _TileColorString {
+	color: String,
+}
+
+impl _TileColorString {
+	fn parse_color(self) -> Result<_Parsed<All>, _Error> {
+		let chars = &mut self.color.chars();
+
+		_Parsed::parse_prefix(chars)?
+			.parse_red(chars)?
+			.parse_blue(chars)?
+			.parse_green(chars)?
+			.parse_alpha(chars)?
+			.parse_end_of_string(chars)
```

Comment:
Why not move `chars` into `_Parsed`? This way we cannot event manipulate it between calling the parser functions.

Reply:
done via the new generic parser